File: volunteer-service/src/main/java/com/nurdor_project/volunteer_service/model/City.java
package com.nurdor_project.volunteer_service.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.util.LinkedHashSet;
import java.util.Set;

@Getter
@Setter
@Entity
@Table(name = "city")
public class City {
    @Id
    @Size(max = 10)
    @Column(name = "zipCode", nullable = false, length = 10)
    private String zipCode;

    @Size(max = 100)
    @NotNull
    @Column(name = "cityName", nullable = false, length = 100)
    private String cityName;

    @OneToMany
    @JoinColumn(name = "nearestCity")
    private Set<Volunteer> volunteers = new LinkedHashSet<>();


}

File: RMA_Project_v2/app/src/main/java/com/example/rma_nurdor_project_v2/model/City.kt
package com.example.rma_nurdor_project_v2.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
data class City(
    @PrimaryKey val zipCode: String,
    val cityName: String
) {
    override fun toString(): String {
        return cityName
    }
}
